### PR TITLE
Changes to CLI to support Axios Proxy, Large FHIR List Downloads, and Downloads of Layouts

### PR DESCRIPTION
**Pull request by taylordeatri**
#### Changed FHIR list to support option to output in a CSV, TSV, etc. format using a json file that maps JmesPaths to column names for building a CSV from the deep json structure.
* Changed to print results as they are received by default (was by an option before.)
* Also added a global --jsonLine print option to support printing the json one resource per line (i.e., without pretty print).

#### Added apps.js to support calling application APIs
* Used to get Configured Layouts for a project.

### CODE REVIEW

File: csv_medadmin.json [hunk 1]
```diff
@@ -0,0 +1,43 @@
+{
```
Comment (mschroering) [inline csv_medadmin.json]: Did you mean to commit this file and csv_patient.json?

Comment (taylordeatri) [reply csv_medadmin.json]: I did but they should probably go to an examples folder - they are sample csv format files for exporting fhir list as --csv <format file>.

Comment (taylordeatri) [reply csv_medadmin.json]: I moved the files to examples/csv_format folder.

**Comment by hemp** [comment]
I think the `package-lock.json` file can be removed to avoid confusion on which package manager is in use (yarn).

File: test/unit/commands/fhir-test.js [hunk 2]
```diff
@@ -61,6 +61,7 @@ test.serial.cb('The "fhir" command should list fhir resources', t => {
 
   yargs.command(list)
     .parse('list Patient --project projectId');
+  t.end();
```
Comment (mschroering) [inline test/unit/commands/fhir-test.js]: I think this will end the test before the verification completes in the callback.

**Comment by mschroering** [comment]
> I think the `package-lock.json` file can be removed to avoid confusion on which package manager is in use (yarn).

This file should still be removed as the project uses  a yarn lock file.

File: test/unit/commands/fhir-test.js (hunk 2 above)
Comment (mschroering) [reply test/unit/commands/fhir-test.js]: You may need to mock out there proxy module here like you did for `api.test.js`.

Comment (taylordeatri) [reply test/unit/commands/fhir-test.js]: Thanks - I'll re-test it with the mock.

**Comment by taylordeatri** [comment]
I tried axios 1.18.0 and get this error when hitting the outbound proxy:

> ./lo fhir list Patient --limit 1
Error: write EPROTO 140631739561856:error:140770FC:SSL routines:SSL23_GET_SERVER_HELLO:unknown protocol:../deps/openssl/openssl/ssl/s23_clnt.c:827:

This appears to be related to this issue which is fixed in 1.19.0:
https://github.com/axios/axios/issues/925

So to enable the proxy it appears we need the 1.19.0-beta release.

**Comment by anthonyroach** [comment]
@taylordeatri I also tried using 1.18.0 with a proxy and got the same error. Looks like using the beta is the only option for now.

**Comment by mschroering** [comment]
> @taylordeatri I also tried using 1.18.0 with a proxy and got the same error. Looks like using the beta is the only option for now.

@anthonyroach you were able to get this to work with the axios beta, right?

`https_proxy=https://45.76.76.19:3128 lo projects list`

@taylordeatri can you use the environment variable approach?  Then we would not need to add the proxy configuration to the setup.

**Comment by anthonyroach** [comment]
@k

> @anthonyroach you were able to get this to work with the axios beta, right?
> 
> `https_proxy=https://45.76.76.19:3128 lo projects list`

Yes I was able to get that working with the axios beta.

**Comment by anthonyroach** [comment]
Oops, didn't mean to close this PR.

**Comment by taylordeatri** [comment]
I'm closing this - tested and confirmed that with the 1.19.0-beta.1 version of axios the property https_proxy DOES work. I'm removing the Proxy configuration code in the changes and will re-submit a new PR.
Thank you Anthony and Mark!

**Pull request merged**